packages/fxa-settings/src/pages/Pair/Index/index.tsx

```diff
@@ -0,0 +1,178 @@
+    // This will just run at page load.
+    if (!isSupported()) {
+      navigate('/pair/unsupported');
+    }
+    if (isDefaultAccount()) {
+      // we have historically needed the "forceView" option to prevent a loop of redirects.
+      navigate('/connect_another_device', { state: { forceView: true } });
+    }
+    if (accountIsNotVerifiedOrHasNoSessionToken()) {
+      navigate('/signin');
+    }
+  }, [navigate]);
+
+  const showQRCode =
+    entryPoint === ENTRYPOINTS.FIREFOX_MENU_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_PREFERENCES_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_SYNCED_TABS_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TABS_SIDEBAR_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_FX_VIEW_ENTRYPOINT ||
+    entryPoint === ENTRYPOINTS.FIREFOX_TOOLBAR_ENTRYPOINT;
+
+  return (
+    <>
+      {showQRCode ? (
+        <CardHeader
+          headingTextFtlId="pair-cad-header"
+          headingText="Connect Firefox on another device"
+        />
+      ) : (
+        <CardHeader
+          headingTextFtlId="pair-sync-header"
+          headingText="Sync Firefox on your phone or tablet"
+        />
+      )}
+
+      <section>
+        {error && (
+          <Banner type={BannerType.error}>
+            <p>{error}</p>
+          </Banner>
+        )}
+        {showQRCode ? (
+          <>
+            <div className="mt-9 mb-5">
+              <FtlMsg id="pair-already-have-firefox-paragraph">
+                <p className="text-sm">
+                  Already have Firefox on a phone or tablet?
+                </p>
+              </FtlMsg>
+            </div>
+            <div className="flex">
+              <FtlMsg id="pair-sync-your-device-button">
+                <button
+                  className="cta-primary cta-xl"
+                  type="button"
+                  onClick={() => {
+                    onSubmit();
+                  }}
+                >
+                  Sync your device
+                </button>
+              </FtlMsg>
+            </div>
+            <FtlMsg id="pair-or-download-subheader">
+              <h2 className="text-lg font-bold mt-10">Or download</h2>
+            </FtlMsg>
+            <FtlMsg
+              id="pair-scan-to-download-message"
+              elems={{
+                linkExternal: (
+                  <LinkExternal
+                    href="https://www.mozilla.org/firefox/mobile/get-app/"
+                    className="link-blue"
+                  >
+                    download link.
+                  </LinkExternal>
+                ),
+              }}
+            >
+              <p className="my-5 px-16 text-sm">
+                Scan to download Firefox for mobile, or send yourself a{' '}
+                <LinkExternal
+                  href="https://www.mozilla.org/firefox/mobile/get-app/"
+                  className="link-blue"
+                >
+                  download link.
+                </LinkExternal>
```

Comment:
Since this looks like a dupe of L119-124, I think a different PR was making this its own little component that it was sharing between the two uses to reduce copy-paste. (But I could be mis-remembering, so I'd have to scan through recent PRs for proof of this half-baked theory).